packages/web3-eth/types/index.d.ts

```diff
@@ -220,3 +222,17 @@ export interface Subscription<T> {
 
     on(type: 'error', handler: (data: Error) => void): void
 }
+
+export interface GetProof {
+    jsonrpc: string;
+    id: number;
+    result: {
+      address: string;
+      accountProof: string[];
+      balance: number | string;
+      codeHash: string;
+      nonce: number | string;
+      storageHash: string;
+      storageProof: [];
```

Comment:
A small detail but I would define it with: ``any[]``

Reply:
```
storageProof: ARRAY - Array of storage-entries as requested. Each entry is a object with these properties:

key: QUANTITY - the requested storage key
value: QUANTITY - the storage value
proof: ARRAY - Array of rlp-serialized MerkleTree-Nodes, starting with the storageHash-Node, following the path of the SHA3 (key) as path.
```

example:

```js
"storageProof": [
      {
        "key": "0x56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421",
        "proof": [
          "0xf90211a...0701bc80",
          "0xf90211a...0d832380"
        ],
        "value": "0x1"
      }
    ]
```

Reply:
Yes, right @joshstevens19. Thanks.

Reply:
@joshstevens19 T[ ] is deprecated. linter says that we should use Array<T> instead.

Reply:
Both do the same thing, which linter is saying that? `Array<T>` and `T[]` should both be supported. Keep it the same `T[]` for now if that is a thing from typescript then we can change it everywhere.

Reply:
could we use:-
```
export interface GetProof {
    jsonrpc: string;
    id: number;
    result: {
      address: string;
      accountProof: string[];
      balance: string;
      codeHash: string;
      nonce: string;
      storageHash: string;
      storageProof: Array<StorageProof>;
    };
}

export interface StorageProof {
    key: string;
    proof: string[];
    value: string;
}
```

Reply:
`Array<StorageProof>` and `StorageProof[]` are exactly the same things. To be consistent i would use `StorageProof[]`

Reply:
Yes, you are right. Typescript supports both,

> which linter is saying that?

can you please check the build pipeline?

Reply:
okay so, for now, I will remain it as `StorageProof[ ].` In future, we can change.

Reply:
TypeScript do not state you should use 1 or the other its up to you. What your seeing is a linter config set up. This error will go away when you turn it into a interface as you shown above. 

We should use `T[]` throughout as it follows consistency and TypeScript say you can use both and `T[]` is defo not `deprecated`

Reply:
okay cool. Thanks.

Reply:
This was the build pipeline error @joshstevens19 
`array-type  Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>' instead.`

Reply:
yep its a bad error its basically saying `create a interface for it` and not do a dynamic object like you did before (and my example shown). That is coming from dtslint tslint - its not a ts `deprecated` warning.

Reply:
Cool. Thanks for clarification @joshstevens19.